.gitlab/kernel_version_testing/system_probe.yml

```
@@ -404,16 +407,30 @@ kernel_matrix_testing_cleanup:
   script:
     - |
       if [ -f $CI_PROJECT_DIR/stack.outputs ]; then
-        INSTANCE_IPS=$(cat $CI_PROJECT_DIR/stack.outputs | grep -e '-instance-ip' | cut -d ' ' -f 2 | tr '\n' ' ' | awk '{$1=$1};1' | tr ' ' ',')
+        INSTANCE_IPS=$(cat $CI_PROJECT_DIR/stack.outputs | grep -e "$ARCH-instance-ip" | cut -d ' ' -f 2 | tr '\n' ' ' | awk '{$1=$1};1' | tr ' ' ',')
         IP_FILTER="Name=private-ip-address,Values=$INSTANCE_IPS"
       else
         echo "No ${CI_PROJECT_DIR}/stack.outputs file found"
         IP_FILTER=""
       fi
     - !reference [.shared_filters_and_queries]
-    - INSTANCE_IDS=$(aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE  $IP_FILTER --output text --query $QUERY_INSTANCE_IDS | tr '\n' ' ')
+    - INSTANCE_IDS=$(aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE $IP_FILTER --output text --query $QUERY_INSTANCE_IDS | tr '\n' ' ')
     - echo $INSTANCE_IDS
     - |
       if [ $(echo $INSTANCE_IDS | wc -w) != "0" ]; then
         aws ec2 terminate-instances --instance-ids $INSTANCE_IDS
       fi
```

Can we use the JSON-based approach we found in ami-builder?
```suggestion
    - aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE $IP_FILTER --output json --query $QUERY_INSTANCE_IDS | tee -a instance.json
    - INSTANCE_ID="$(jq -r '.Instances[0].InstanceId' < instance.json)"
    - echo ${INSTANCE_ID}
    - |
      if [[ "${INSTANCE_ID}" != "" ]]; then
        aws ec2 terminate-instances --instance-ids "${INSTANCE_ID}"
      fi
```